xbmc/peripherals/devices/PeripheralCecAdapter.cpp

```diff
@@ -1856,3 +1859,41 @@ bool CPeripheralCecAdapter::ToggleDeviceState(CecStateChange mode /*= STATE_SWIT
 
   return false;
 }
+
+CPeripheralCecAdapter::CecPowerState CPeripheralCecAdapter::GetConnectedDevicesPowerState() const
+{
+  cec_logical_addresses controlledAdresses = m_cecAdapter->GetLogicalAddresses();
+
+  uint8_t standingByDevices = 0;
+  uint8_t activeDevices = 0;
+  bool hasChangingPowerDevices = false;
+
+  for (uint8_t i = 0; i < 16; ++i)
+  {
+    if (controlledAdresses[i])
+    {
+      const cec_power_status devicePowerStatus =
+          m_cecAdapter->GetDevicePowerStatus(static_cast<cec_logical_address>(i));
```

Comment:
GetDevicePowerStatus() shouldn't return anything valid for address 15, the braodcast address.